src/software/new_geom/util/closest_point.h

```diff
@@ -12,3 +13,17 @@
  */
 Point closestPointOnLine(const Point &p, const Line &l);
 Point closestPointOnLine(const Line &l, const Point &p);
+
+/**
+ * Finds the Point on line segment closest to point.
+ *
+ * @param centre the point.
+ *
+ * @param segA one end of the line segment.
+ *
+ * @param segB the other end of the line segment.
+ *
```

Comment:
should take a segment

Reply:
Added comments for `closestPointOnSeg` taking in point and segment.